### PR TITLE
Report JS errors to sentry

### PR DESCRIPTION
**Pull request by dannyroberts**
##### SUMMARY
I figure we should be getting these in sentry, where we can actually see the useful information and trace it back to particular urls, users, etc.

### CODE REVIEW

File: corehq/apps/hqwebapp/views.py [hunk 1]
```diff
@@ -613,13 +613,35 @@ def jserror(request):
             browser_version = parsed_agent['browser'].get('version', TAG_UNKNOWN)
             browser_name = parsed_agent['browser'].get('name', TAG_UNKNOWN)
 
+    url = request.POST.get('page', None)
+    domain = '_unknown'
+    if url:
+        parts = url.split('/')
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/views.py:619]: Would `get_domain_from_url` work here?

**Review by orangejenny — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

File: corehq/apps/hqwebapp/views.py (hunk 1 above)
Comment (dannyroberts) [reply corehq/apps/hqwebapp/views.py:619]: Ah, that's a good point. `get_domain_from_url` (which I didn't know about before) appears to take as input the path that comes after the <protocol>://<domain> section of the url (i.e. `/a/{domain}/...`, whereas `url` here includes the protocol and domain part of the url. So I can't pass it directly in, but I can pass it through a real URL parser to get the path, and then pass the path to this function. That's better than my makeshift parsing here

File: corehq/apps/hqwebapp/views.py [hunk 2]
```diff
@@ -613,13 +613,35 @@ def jserror(request):
             browser_version = parsed_agent['browser'].get('version', TAG_UNKNOWN)
             browser_name = parsed_agent['browser'].get('name', TAG_UNKNOWN)
 
+    url = request.POST.get('page', None)
+    domain = '_unknown'
+    if url:
+        path = urlparse(url).path
+        if path:
+            domain = get_domain_from_url(path)
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/views.py:621]: Could this switch `domain` from `_unknown` to `None` if it isn't a domain-specific url?

Comment (dannyroberts) [reply corehq/apps/hqwebapp/views.py:621]: Oh, yes thanks

**Review by orangejenny — APPROVED** [review]

**Comment by snopoke** [comment]
@dannyroberts I'm curious why we took this approach vs using Sentry's built in JS error reporting capabilities: https://sentry.io/for/javascript/

**Comment by dannyroberts** [comment]
I honestly don't remember. One possibility is that reporting from the client side requires exposing a key anyone could use to spam our sentry, though I'm not sure how big a concern that is.

**Pull request merged**